### PR TITLE
fix bug when hero rebirth would lose skin

### PR DESCRIPTION
**Pull request by evalexp**

### CODE REVIEW

**Comment by kahuixiansheng** [comment]
> _ No description provided. _

Thinks for your update

File: R3nzSkin/offsets.hpp [hunk 1]
```diff
@@ -15,7 +15,9 @@ namespace offsets {
 		};
 		enum {
 			Team = 0x34,
-			Name = 0x54
+			Name = 0x54,
+			Health = 0xE7C,
+			MaxHealth = 0xE8C
```
Comment (cjd0901) [inline R3nzSkin/offsets.hpp:20]: I‘m wondering how you know these enum types, use loop violence call?

Comment (evalexp) [reply R3nzSkin/offsets.hpp:20]:
```cpp
#pragma once
 
namespace offsets {
 
	DWORD baseAdrr = (DWORD)GetModuleHandle(NULL);
 
	namespace D3DX {
		DWORD Instance = 0x31A13E4; //13.5 //0x319DE74;//13.4 //0x31A5370; // x "Failed to create DX11Query" -> first instruction -> mov eax, dword_30B392C
		DWORD oHwnd = 0x158; 
		DWORD WindowSizeX = 0x12c;
		DWORD WindowSizeY = 0x130;
 
		DWORD Renderer = 0x319E770;// 13.5 //0x319B280;//13.4 //0x31A28B8; // V //57 8B AE ? ? ? ? E8 ? ? ? ? //  A1 ?? ?? ?? ?? 56 57 BF ?? ?? ?? ?? 8B
		DWORD RendererWidth = 0x8;
		DWORD RendererHeight = 0xc;
		DWORD ViewMatrix = 0x3198DB8; // 13.5 //0x31958E8; //13.4 //0x319CF18;  //   V //B9 ? ? ? ? E8 ? ? ? ? B9 ? ? ? ? E8 ? ? ? ? B9 ? ? ? ? 
 
	}
	 
	namespace Chat {
		DWORD Client = 0x3172B90; // 13.5 // 0x316F79C;//13.4 //0x31768A8; //x //  74 1E 8B 09 > jump to loc first dword //8B 0D ? ? ? ? 8A D8 85 C9  // C7 46 ?? ?? ?? ?? ?? 8B CE 5E 5D C7 05 ?? ?? ?? ?? ?? ?? ?? ?? 5B E9 ?? ?? ?? ?? E8 ?? ?? ?? ?? PRIMA DWORD SOTTO // 8B 0D ? ? ? ? 8A D8 85 C9 //8B 0D ? ? ? ? 8A D8 85 C9
		DWORD IsOpen = 0x814;
	}
 
 
 
	DWORD HudInstance = 0x18D4A3C; // 13.5 //0x18D1A5C; //13.4 //0x18D9AD4; //v  8B 0D ?? ?? ?? ?? FF 77 08 8B 49 14 // 8B ? ? ? ? ? 6A 00 8B ? ? E8 ? ? ? ? B0 01 ?
 
	DWORD ZoomClass = 0x316A6D8; // 13.5 //0x31676E8; //13.4 //0x316F458; // v  // A3 ? ? ? ? 83 F9 10 72 32 // 8D 4E 70 C7 05 ? ? ? ? ? ? ? ?  dword sotto
	DWORD ZoomAmount = 0x28; //DA CHECKARE
 
	DWORD GameTime = 0x316B268; // 13.5 // 0x31680E4; //13.4 //0x316FDE4; // v // F3 0F 11 05 ? ? ? ? 8B 49
 
 
	namespace Manager {
		DWORD Objects = 0x18D4A34; // 13.5 //0x18D1A54; //13,4 //0x18D9ACC; // v 13.3// 8B 0D ?? ?? ?? ?? 89 7C 24 14    //   mov     ecx, objectManager
		DWORD HeroList = 0x18D4AD4; // 13.5 //0x18D1AF4; //13,4 //0x18D9B6C;		// v // 89 44 24 18 A1 ? ? ? ? 53 55				---> first address;       mov     eax, dword_
		DWORD TurretList = 0x3169D6C; // 13.5 //0x3166D74; //13,4 //0x316EAC4;  //  v	"towercautionRing_Minimal_Fountain_White"	---> 2 above;		 mov     dword_3325A48, esi
		DWORD MissileList = 0x3172B3C; // 13.5 //0x316F748; //13,4 //0x3176830; // v    // 84 C9 8D 85 ? ? ? ?  // 8B 0D ?? ?? ?? ?? E8 ?? ?? ?? ?? 8B 30 8B 36 // E8 ? ? ? ? 8B 0D ? ? ? ? 8D 44 24 1C 50 DWORD SOTTO
	}
 
 
	namespace Missiles {
		DWORD StartPos = 0x2E0;
		DWORD EndPos = 0x2EC;
		DWORD SourceIndex = 0x2C4;
		DWORD DestinationIndex = 0x31c;
		DWORD SpellInfo = 0x0260; //260
		DWORD SpellDataMissileName = 0x18;
	}
 
	DWORD SpellBook = 0x2250;
	namespace SpeelSlots {
		DWORD Instance = 0x4C8;
		DWORD Level = 0x1c;
		DWORD Time = 0x24;
		DWORD Charges = 0x58;
		DWORD SpellInfo = 0x120;
		DWORD SpellInfoSpellData = 0x40;
		DWORD SpellDataName = 0x6C;
	}
	namespace SpellEntry {
		DWORD Info = 0x8;
		DWORD Slot = 0xC;
		DWORD StartPos = 0x88;
		DWORD EndPos = 0x94;
		DWORD IsBasicA = 0xeC;
 
		namespace SpellInfo {
			DWORD Name = 0x6c;
			DWORD Data = 0x40;
 
			namespace SpellData{
				DWORD Range = 0x3c0;
				DWORD Radius = 0x3f8;
				DWORD Speed = 0x44c;
				DWORD Width = 0x474;
			}
		}
 
	}
 
	DWORD MapInstance = 0x18D4D88; // 13.5 //0x18D1DA4; //13.4 //0x18D9E1C; // 13.3 FF 90 ? ? ? ? 8B 0D ? ? ? ? 89 88 ? ? ? ? 
	DWORD MouseScreen = 0x2522170; // 13.5 //  8B 0D ? ? ? ? 8D 44 24 34 50 8D 44 24 30
 
	namespace Functions {
 
		DWORD GameUpdate = 0x324FA0; //13.4 //0x2E7FC0; // CALLS OBJECTAANGER UPDATE -> Very slow update
		DWORD ObjectManagerUpdate = 0x2AC5D0; // 13.4 //0x295E00;//E8 ? ? ? ? E8 ? ? ? ? 6A 05
 
		DWORD IsHero = 0x18D2F0; // 13.5 //0x18C2F0;//13.4//0x187F30; //13.3 //E8 ? ? ? ? 83 C4 04 5B 84 C0
		DWORD IsMinion = 0x19F8C0; //13.1 //800h 
		DWORD IsTurret = 0x19F9B0; //13.1 //2000h
		DWORD IsMissile = 0x19F8E0; //13.1 //8000h 
		DWORD IsInhibitor = 0x19F820; //13.1 //2th
		DWORD IsNexus = 0x19F860; //13.1 //3th
		DWORD IsAlive = 0x17FEA0; // 13.5 // 0x17F260; //13.4 //0x17AF50; // 13.3 E8 ? ? ? ? 84 C0 74 2A 8D 8F ? ? ? ? 
		DWORD IsNotWall = 0xA3F000;// 13.1
 
		DWORD IssueOrder = 0x167340;// x 13.1
		DWORD NewIssueOrder = 0x66BA80; // 13.5 // 0x676B00; //13.4 //0x68F900; // 13.3 //E8 ? ? ? ? 84 C0 74 13 C7 46 ? ? ? ? ?
		DWORD NewCastSpell = 0x651C80; // 13.5 //0x65C7D0;  //13.4   //0x6715E0;;// v 13.3
		DWORD GetSpellState = 0x523610; // 13,5 //0x532690; //13.4 //0x53A620;//v  x13.3 // 83 EC 14 8B 44 24 1C 55
		DWORD GetBasicAttack = 0x154CE0; //13,4 //0x1519E0; // v 13.3 // E8 ? ? ? ? 83 38 00 74 4B 
		DWORD SpoofGadget = 0xF7F0E3;// v 13.4 // ff 23 jmp dword ptr [ebx]
 
		DWORD GetStatTotal = 0x1C23B0;
 
		DWORD CompareFlags = 0x15F750; //13.4 //0x15C910; //13.3 E8 ? ? ? ? 84 C0 0F 84 ? ? ? ? 8A 43 DD
		DWORD OnCreateObject = 0x297020;// 13.4 //0x293A20;	//E8 ? ? ? ? 8B 1D ? ? ? ? 8D 44 24 34
		DWORD OnDeleteObject = 0x2A5250; //13,4 //0x282370;	//E8 ? ? ? ? 8B 4E 20 85 C9 74 0F
 
		DWORD WorldToScreen = 0xAAD8F0;  // x	85 C0 74 0C 8B 40 04 //  primo sub soprra 
		DWORD PrintChat = 0x5EDE10; // 13.5 //0x5FF9A0; //13.4 //0x607640;  // v game_chat_command_help" SOPRA PRIMA SUB // "game_chat_known_commands" //55 8B EC 83 EC 40 A1 ? ? ? ? 33 C5 89 45 FC 8B C1 // hud_Chat_bounty_ping // E8 ? ? ? ? 8B 4C 24 20 C6 47 0D 01 
		DWORD SendChat = 0x689880;  //A1 ? ? ? ? 56 6A FF.
		
		DWORD GetHeightAtPos = 0xA4B8A0; // 13.5 //0xA47C90; //13.4 //0xA4CF00; // 13.3 //E8 ? ? ? ? 8B 44 24 0C D9 18 
 
		DWORD GetAttackDelay = 0x2A3B60; // 13.5 // 0x2A4720; //13.4 //0x2A0860;	// 	v	8B 44 24 04 51 F3 // 33 C9 F3 0F 10 40 ?  10E52A0  // EB 03 8B 47 08 - second sub_ //E8 ? ? ? ? D8 44 24 14 83 C4 04 ---------------------------------------------------------- D9 9E ? ? ? ? 57 E8 ? ? ? ? SUB SOTTO 
		DWORD GetAttackCastDelay = 0x2A3A60; // 13.5 //0x2A4620; //13.4 //0x2A0760;	// v	83 EC 0C 53 8B 5C 24 14 8B CB 56 57 8B 03 FF 90 // fn // 83 EC 0C 53 8B 5C 24 14 8B CB 56 57 8B 03 FF 90 // EB 03 8B 47 08 - first sub_  E8 ? ? ? ? D9 9E ? ? ? ? 57----------- D9 9E ? ? ? ? 57 E8 ? ? ? ? SUB SOPRA
 
		DWORD GetBaseDrawPosition = 0x153D00; // v // E8 ?? ?? ?? ?? EB 07 8B 01
		DWORD GetHealthBarPosition = 0x61A1E0; // 13.5	 //0x627220; //13.4 //0x62D7C0;  // v // F3 0F 11 44 24 ? E8 ? ? ? ? F3 0F 10 44 24 ? 51 F3 0F 11 04 24 8B CF 50 E8 ? ? ? ? 5F // 85 C0 74 0C 8B 40 04 inside fu nction
 
		DWORD GetRespawnTimer = 0x1600A0; // x 83 EC 08 56 8B F1 57 8D 8E ?? ?? 00 00 E8 VT228?
 
		DWORD OnProcessSpellCast = 0x54AA30;	//E8 ? ? ? ? 85 C0 0F 94 44 24 ?
		DWORD OnBuff = 0x541E50;// 13.4 //0x549B00;	// 53 55 8B 6C 24 0C 8B D9 55
	}
 
	DWORD UnderMouseObject = 0x252230C; // 13.5 //0x251F3AC; //13.4 //0x252741C; // x ADD 0XC 89 0D ? ? ? ? C7 41 ? ? ? ? ? C7 41 ? ? ? ? ?
	DWORD LocalPlayer = 0x3172ABC; // 13.5 //0x316F708; //13.4 //0x31767BC; //	v		8B 0D ? ? ? ? 81 C1 ? ? ? ? 8B 01 FF 50 10 39 43 34
	namespace GObject {
 
		DWORD Index = 0x0008;
		DWORD Team = 0x0034;
		DWORD Name = 0x54;//
		DWORD NetworkID = 0xB4;
		DWORD Pos = 0x01DC;//
		DWORD Visibility = 0x0274;
		DWORD Health = 0xE7C;//
		DWORD MaxHealth = 0xE8C;
		DWORD Mana = 0x29C;
		DWORD MaxMana = 0x2AC;
		DWORD Invulnerable = 0x3D4;
		DWORD Targetable = 0x0D04;
		///DWORD RecallState = 0xD90;
		DWORD CombatType = 0x2268; //OUTDATED
		DWORD AtkRange = 0x13a4; 
		DWORD ChampionName = 0x2cf0;
		DWORD MovementSpeed = 0x139c; //
 
	}
}
 
#define RVA(address) (offsets::baseAdrr + address)
```

ref: https://www.unknowncheats.me/forum/league-of-legends/310587-league-legends-reversal-structs-offsets-433.html

Comment (cjd0901) [reply R3nzSkin/offsets.hpp:20]: thx

**Comment by hotline1337** [comment]
Thank you for the pull request, although i cannot approve it due to its heavily non optimized code structure.

**Pull request closed without merging**